src/Controls/src/Core/Platform/Android/BottomNavigationViewUtils.cs

```diff
@@ -132,9 +132,10 @@ internal static async Task SetMenuItemIcon(IMenuItem menuItem, ImageSource sourc
 			if (source is null)
 				return;
 
-			var services = context.Services;
-			var provider = services.GetRequiredService<IImageSourceServiceProvider>();
-			var imageSourceService = provider.GetRequiredImageSourceService(source);
+			if (context.Services is not IKeyedServiceProvider provider)
+				throw new InvalidOperationException($"Context services {nameof(context.Services)} does not implement {nameof(IKeyedServiceProvider)}");
```

Comment:
@PureWeen can/should we change `context.Services` to an `IKeyedServiceProvider`?  (it extends `IServiceProvider`) Then we wouldn't need to cast/throw here?

https://github.com/dotnet/runtime/blob/0823c5c2814ffad5718f06715899ef91aa84aae2/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/IKeyedServiceProvider.cs#L12

The cast/throw occurs a couple places in this PR.